src/ast/query.rs

```diff
@@ -35,8 +35,8 @@ pub struct Query {
 
 impl fmt::Display for Query {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        if !self.ctes.is_empty() {
-            write!(f, "WITH {} ", display_comma_separated(&self.ctes))?;
+        if self.with.is_some() {
+            write!(f, "{}", self.with.as_ref().unwrap())?;
```

Comment:
`if let Some(ref with) = self.with`? Also: I think it makes more sense to print the space between the CTEs and the main query here, as part of `Query` formatting, rather than trailing make `With` end with a trailing space.